pulp_rpm/tests/functional/api/test_publish.py

```diff
@@ -333,3 +342,141 @@ def _get_updateinfo_xml_path(root_elem):
+        policy = kwargs.get("policy", "immediate")
+        # 1. create repo and remote
+        repo = self.repo_api.create(gen_repo())
+        self.addCleanup(self.repo_api.delete, repo.pulp_href)
+
+        body = gen_rpm_remote(policy=policy)
+        remote = self.remote_api.create(body)
+        self.addCleanup(self.remote_api.delete, remote.pulp_href)
+
+        # 2. Sync it
+        repository_sync_data = RpmRepositorySyncURL(remote=remote.pulp_href)
+        sync_response = self.repo_api.sync(repo.pulp_href, repository_sync_data)
+        monitor_task(sync_response.task)
+
+        # 3. Publish and distribute
+        publish_data = RpmRpmPublication(repository=repo.pulp_href)
+        if package_checksum_type:
+            publish_data.package_checksum_type = package_checksum_type
+        if metadata_checksum_type:
+            publish_data.metadata_checksum_type = metadata_checksum_type
+        publish_response = self.publications.create(publish_data)
+        created_resources = monitor_task(publish_response.task)
+        publication_href = created_resources[0]
+        self.addCleanup(self.publications.delete, publication_href)
+
+        body = gen_distribution()
+        body["publication"] = publication_href
+        distribution_response = self.distributions.create(body)
+        created_resources = monitor_task(distribution_response.task)
+        distribution = self.distributions.read(created_resources[0])
+        self.addCleanup(self.distributions.delete, distribution.pulp_href)
+
+        repomd = ElementTree.fromstring(
+            http_get(
+                os.path.join(distribution.base_url, 'repodata/repomd.xml')
+            )
+        )
+
+        data_xpath = '{{{}}}data'.format(RPM_NAMESPACES['metadata/repo'])
+        data_elems = [elem for elem in repomd.findall(data_xpath)]
+
+        repomd_checksum_types = {}
+        primary_checksum_types = {}
+        checksum_xpath = '{{{}}}checksum'.format(RPM_NAMESPACES['metadata/repo'])
+        for data_elem in data_elems:
+            checksum_type = data_elem.find(checksum_xpath).get('type')
+            repomd_checksum_types[data_elem.get("type")] = checksum_type
+            if data_elem.get("type") == "primary":
+                location_xpath = '{{{}}}location'.format(RPM_NAMESPACES['metadata/repo'])
+                primary_href = data_elem.find(location_xpath).get('href')
+                primary = ElementTree.fromstring(
+                    read_xml_gz(http_get(
+                        os.path.join(distribution.base_url, primary_href)
+                    ))
+                )
+                package_checksum_xpath = '{{{}}}checksum'.format(RPM_NAMESPACES['metadata/common'])
+                package_xpath = '{{{}}}package'.format(RPM_NAMESPACES['metadata/common'])
+                package_elems = [elem for elem in primary.findall(package_xpath)]
+                for package_elem in package_elems:
```

Comment:
there is no need to iterate though the whole packages, imagine you have 32 rpms you will iterate though all 32 elements and you will keep re-assigning to the same dict key the pkg_checksum_type.
It is enough to take 1 element and extract the checksumtype